2019-05-14-armagaddon/heka/generate_batches

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env python
+# generate s3api delete object batches
+# https://docs.aws.amazon.com/cli/latest/reference/s3api/delete-objects.html
+import sys
+import json
+j = {"Objects": [{"Key": "test1.txt"}],"Quiet": False}
+
+
+def chunks(l, n):
+    """Yield successive n-sized chunks from l."""
+    for i in xrange(0, len(l), n):
+        yield l[i:i + n]
+
+
+f = open(sys.argv[1], 'r').readlines()
+for i, chunk in enumerate(chunks(f, 1000)):
+    objects = []
+    for line in chunk:
+        objects.append({"Key": (line.rstrip().split(' ')[-1])})
+    j["Objects"] = objects
+    with open("batch{}.json".format(i), "w") as f:
+        f.write(json.dumps(j))
```

Comment:
this f is shadowing the one used to create the initial for loop, it should be fine, but I think it's worth renaming the variable for safety